crates/nu-command/src/strings/str_/expand.rs

```diff
@@ -0,0 +1,274 @@
+                expansions.push(combination);
+            }
+        }
+    } else {
+        return None;
+    }
+    if expansions.is_empty() {
+        None
+    } else {
+        Some(expansions)
+    }
+}
+
+fn combine(prefix: &str, content: &str, postfix: &str) -> String {
+    format!("{}{}{}", prefix, content, postfix)
+}
+
+fn split(content: impl ToString) -> Option<Vec<String>> {
+    let content = content.to_string();
+    if content.is_empty() {
+        return None;
+    }
+    let mut pieces: Vec<String> = Vec::new();
+    let mut count = (0, 0); // right, left / open, close
+    let mut piece = String::new();
+    for c in content.chars() {
+        match c {
+            '{' | '}' => {
+                piece.push(c);
+                if c == '{' {
+                    count.0 += 1;
+                } else {
+                    count.1 += 1;
+                }
+            }
+            ',' if count.0 == count.1 => {
+                pieces.push(piece.clone());
+                piece.clear();
+            }
+            _ => piece.push(c),
+        }
+    }
+    if !piece.is_empty() {
+        pieces.push(piece);
+    }
+    if pieces.is_empty() {
+        None
+    } else {
+        Some(pieces)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_expand_simple() {
+        let input = "c{d,e}f";
+        let expected_output: Vec<String> = vec!["cdf".into(), "cef".into()];
+        assert_eq!(expand(input), Some(expected_output));
+    }
+
+    #[test]
+    fn test_expand_recursive1() {
+        let input = "a{b,c{d,e}f,g}h";
+        let output: Vec<String> = vec!["abh".into(), "acdfh".into(), "acefh".into(), "agh".into()];
+        assert_eq!(expand(input), Some(output));
+    }
+
+    #[test]
+    fn test_expand_recursive2() {
+        let input = "a{b,c{d{1,2},e}f,g}h";
+        let output: Vec<String> = vec![
+            "abh".into(),
+            "acd1fh".into(),
+            "acd2fh".into(),
+            "acefh".into(),
+            "agh".into(),
+        ];
+        assert_eq!(expand(input), Some(output));
+    }
+
+    #[test]
+    fn test_split_complex1() {
+        let input = "b,c{d,e}f,g";
+        let output: Vec<String> = vec!["b".into(), "c{d,e}f".into(), "g".into()];
+        assert_eq!(split(input), Some(output));
+    }
+
+    #[test]
+    fn test_split_complex2() {
+        let input = "a,b,c,d{e,f},g{h,i,j},k";
+        let output: Vec<String> = vec![
+            "a".into(),
+            "b".into(),
+            "c".into(),
+            "d{e,f}".into(),
+            "g{h,i,j}".into(),
+            "k".into(),
+        ];
+        assert_eq!(split(input), Some(output));
+    }
+
+    #[test]
+    fn test_basic_brace_expansion() {
+        let input = "{apple,banana,cherry}";
+        let expected_output: Vec<String> = vec!["apple".into(), "banana".into(), "cherry".into()];
+        assert_eq!(expand(&input), Some(expected_output))
+    }
+}
```

Comment:
these tests look good but this is not really how tests should be written in the `nushell` source :relieved: 

1. in order to evaluate and test the examples, you should use the same pattern as in the other command files,
```rust
#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn test_examples() {
        use crate::test_examples;

        test_examples(SubCommand {})
    }
}
```
2. to run tests on the command which are not examples, you should not try to test the Rust functions directly but rather how they behave in the shell
here is an example of such a test
```rust
#[test]
fn upsert_uses_enumerate_index_inserting() {
    let actual = nu!(
        cwd: ".", pipeline(
        r#"[[a]; [7] [6]] | enumerate | upsert b {|el| $el.index + 1 + $el.item.a } | flatten | to nuon"#
    ));

    assert_eq!(actual.out, "[[index, a, b]; [0, 7, 8], [1, 6, 8]]");
}
```
you see that we do not call any Rust function but rather directly write a Nushell pipeline / script to test the commands.
then it's `nu!` and  `actual` which do the job :yum: